### DISCUSSION TITLE
single input parameter syntax for lambda functions

### DISCUSSION THREAD

**Discussion by xp44mm** [fsharp/fslang-suggestions#1190] [General]
Since 2016, there have been many requirements for lambda similar to fat arrows. The fsharp team seem to have accepted some schemes, but no one came down as the stairs rang.

 - https://github.com/fsharp/fslang-suggestions/issues/634
 - https://github.com/fsharp/fslang-suggestions/issues/384
 - https://github.com/fsharp/fslang-suggestions/issues/168
 - https://github.com/fsharp/fslang-suggestions/issues/506

How feasible is we to implement some simple fat arrow head syntax first? What we expect is the following:

```F#
let f = _ => 3    ; fun _ -> 3
let g = () => 3   ; fun () -> 3
let h = x => x    ; fun x -> x

```
nested patterns may not be used in this syntax. Because if the head is too long, too complex, we won't mind entering a more `fun` keyword precede the input parameter pattern. Other case arrow head as complex pattern matching delay in the future. if it is needed!

```F#
(x) => x
(x:string) => x
(x,y) => x,y
(x:string,y:string) => x,y
..
```

**Comment by abelbraaksma** [comment]
This has been proposed and ultimately dismissed in the threads you’ve linked. `=>` is currently a valid operator (your `g` and `h` functions are currently valid syntax). Other downsides are that the syntax you propose cannot easily be made unambiguous, esp since there’s a proposal to repurpose `_`, which makes lexing and parsing difficult if not impossible.

See the linked threads for the preferred ways forward of making ‘fun’ optional, esp Don Syme’s proposal.
